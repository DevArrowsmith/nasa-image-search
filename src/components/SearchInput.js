import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import getImages from '../requests/getImages'


const StyledSearchFrame = styled.div`
  height: 30px;
  background: #222222;
  border-radius: 5px;
  border: 2px solid maroon;
  margin: 0 5px;
  padding: 0 5px;
  display: flex;
  flex-flow: row nowrap;
  justify-content: space-between;
  align-items: center;
`

const StyledSearchInput = styled.input`
  background: transparent;
  border: none;
  margin: 0;
  padding: 0;
  color: #F6F4F0;
  font-size: 1em;
  font-family: 'Roboto', Arial, Helvetica, sans-serif;
`
const StyledSearchIcon = styled.img`
  height: 18px;
  width: auto;
  margin: 0 0 0 5px;
`

const SearchInput = ({ setSearchResults, setResultsLoading }) => {

  const [searchState, setSearchState] = useState("");

  const performSearch = async () => {
    setResultsLoading(true);
    if (!searchState) {
      setSearchResults({noSearchTerm: true})
    } else {
      setSearchResults(await getImages(searchState));
    };
    setResultsLoading(false);
  };
  
  const submitSearch = (event) => {
    if (event.key === 'Enter' || event.target.alt === "Click here to search") {
      performSearch();
    }
  };

  return (
    <>
      <StyledSearchFrame>
        <StyledSearchInput
          type="search"
          placeholder="Search"
          onChange={e => setSearchState(e.target.value)}
          onKeyDown={submitSearch}
        />
        <StyledSearchIcon 
          src="./images/search.png"
          alt="Click here to search"
          onClick={submitSearch}
        />
      </StyledSearchFrame>
    </>
  )
};

SearchInput.propTypes = {
  setSearchResults: PropTypes.func,
  setResultsLoading: PropTypes.func
};

export default SearchInput;