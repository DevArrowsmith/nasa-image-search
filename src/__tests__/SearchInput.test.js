import React from 'react';
import { render } from '@testing-library/react';
import SearchInput from '../components/SearchInput';

test('component SearchInput renders', () => {
  const { asFragment } = render(<SearchInput/>);
  const component = asFragment();
  expect(component).toMatchSnapshot();
});
