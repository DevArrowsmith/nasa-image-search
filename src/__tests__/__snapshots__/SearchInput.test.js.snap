// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component SearchInput renders 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm iGGLME"
  >
    <input
      class="sc-AxirZ jLIEBi"
      placeholder="Search"
      type="search"
    />
    <img
      alt="Click here to search"
      class="sc-AxiKw cjWnXD"
      src="./images/search.png"
    />
  </div>
</DocumentFragment>
`;
